chat.go

```diff
@@ -0,0 +1,108 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/ably/ably-go/ably"
+	"github.com/ably/ably-go/ably/proto"
+)
+
+func main() {
+	fmt.Println("Type your clientID")
+	reader := bufio.NewReader(os.Stdin)
+	clientId, _ := reader.ReadString('\n')
+	clientId = strings.Replace(clientId, "\n", "", -1)
+
+	opts := &ably.ClientOptions{
+		AuthOptions: ably.AuthOptions{
+			// If you have an Ably account, you can find
+			// your API key at https://www.ably.io/accounts/any/apps/any/app_keys
+			Key: "INSERT_YOUR_API_KEY_HERE",
+		},
+		ClientID: clientId,
+		// NoEcho:   true, // Uncomment to stop messages you send from being sent back
+	}
+
+	fmt.Println("You can now send messages!")
+
+	// Connect to Ably using the API key and ClientID specified above
+	client, err := ably.NewRealtimeClient(opts)
+	if err != nil {
+		panic(err)
+	}
+
+	// Connect to the Ably Channel with name 'chat'
+	channel := client.Channels.Get("chat")
+
+	// Enter the Presence set of the channel
+	channel.Presence.Enter("")
+
+	getHistory(*channel)
+
+	go subscribeToPresence(*channel)
```

Comment:
Minor: You also don't need separate go routines here for presence and channels you can do
``` go
sub, err := channel.Subscribe()
if err != nil {
    return fmt.Errorf("subscribing to channel: %w"; err)
}
pSub, err := channel.Presence.Subscribe()
if err != nil {
    return fmt.Errorf("subscribing to presence: %w"; err)
}
select {
case msg := <-sub.MessageChannel():
    ...
case msg := <- pSub.PresenceChannel():
    ....
}
```